awssqs/src/main/java/zipkin/reporter/awssqs/AwsBufferedSqsSender.java

```diff
@@ -0,0 +1,235 @@
+        return CheckResult.OK;
+      } else {
+        return CheckResult.failed(new IllegalStateException(String.format("queue url not found %s", queueUrl())));
+      }
+    } catch (Exception e) {
+      return CheckResult.failed(e);
+    }
+  }
+
+  @Override protected AmazonSQSBufferedAsyncClient compute() {
+    // bound the executor so that we get consistent performance
+    ThreadPoolExecutor dispatchExecutor =
+        new ThreadPoolExecutor(0, maxRequests(), 60, TimeUnit.SECONDS,
+            new ArrayBlockingQueue<>(maxQueuedRequests()),
+            threadFactory("AwsBufferedSqsSender Dispatcher", false));
+
+    AWSCredentialsProvider credentialsProvider =
+        credentialsProvider(awsAccessKeyId(), awsSecretAccessKey(), awsServiceId(), awsRoleName());
+
+    return new AmazonSQSBufferedAsyncClient(
+        new AmazonSQSAsyncClient(credentialsProvider, dispatchExecutor));
+  }
+
+  @Override public int messageSizeInBytes(List<byte[]> encodedSpans) {
+    return encoding().listSizeInBytes(encodedSpans);
+  }
+
+  @Override public void sendSpans(List<byte[]> encodedSpans, Callback callback) {
+    if (closeCalled.get()) throw new IllegalStateException("closed");
+
+    checkNotNull(encodedSpans, "list of encoded spans must not be null");
+
+    for (byte[] span : encodedSpans) {
+      get().sendMessageAsync(new SendMessageRequest(queueUrl(), "z")
+              .addMessageAttributesEntry("z", new MessageAttributeValue()
+                  .withDataType("Binary." + encoding().name())
+                  .withBinaryValue(ByteBuffer.wrap(span))), new AsyncHandlerAdapter(callback));
+    }
+
+  }
+
+  @Override public void close() {
+    if (!closeCalled.getAndSet(true)) {
+      AmazonSQSBufferedAsyncClient maybeNull = maybeNull();
+      if (maybeNull != null) maybeNull.shutdown();
+    }
+  }
+
+  private static AWSCredentialsProvider credentialsProvider(
+      String accessKeyId, String secretAccessKey, String serviceId, String roleName) {
+
+    AWSCredentials credentials;
+    if (accessKeyId == null || secretAccessKey == null) {
+      credentials = new AnonymousAWSCredentials();
+    } else {
+      credentials = new BasicAWSCredentials(accessKeyId, secretAccessKey);
+    }
+
+    AWSCredentialsProviderChain credentialsProviderChain =
+        new CustomAWSCredentialsProviderChain(credentials);
+    AWSCredentialsProvider credentialsProvider = credentialsProviderChain;
+    if (roleName != null) {
+      String roleArn = String.format("arn:aws:iam::%s:role/%s", serviceId, roleName);
+      String roleSessionName = "zipkin" + Math.random();
+      credentials = credentialsProviderChain.getCredentials();
+      credentialsProvider =
+          new STSAssumeRoleSessionCredentialsProvider(credentials, roleArn, roleSessionName);
+    }
+
+    return credentialsProvider;
+  }
+
+  private static ThreadFactory threadFactory(final String name, final boolean daemon) {
+    return (Runnable r) -> {
```

Comment:
The zipkin-reporter-java project uses the Maven Retrolambda plugin.  https://github.com/openzipkin/zipkin-reporter-java/blob/master/pom.xml#L208  I've never used this but my understanding is that it converts lambda expressions to Java6 compatible byte code.